cdap-app-fabric/src/main/java/io/cdap/cdap/app/guice/AppFabricServiceRuntimeModule.java

```diff
@@ -294,11 +297,23 @@ protected void configure() {
       bind(OwnerAdmin.class).to(DefaultOwnerAdmin.class);
       bind(CoreSchedulerService.class).in(Scopes.SINGLETON);
       bind(Scheduler.class).to(CoreSchedulerService.class);
-      bind(ArtifactRepository.class)
-        .annotatedWith(Names.named(NOAUTH_ARTIFACT_REPO))
-        .to(DefaultArtifactRepository.class)
-        .in(Scopes.SINGLETON);
-      bind(ArtifactRepository.class).to(AuthorizationArtifactRepository.class).in(Scopes.SINGLETON);
+      install(new PrivateModule() {
+        @Override
+        protected void configure() {
+          // ArtifactRepositoryReader is required by DefaultArtifactRepository.
+          // Keep ArtifactRepositoryReader private to minimize the scope of the binding visibility.
+          bind(ArtifactRepositoryReader.class).to(LocalArtifactRepositoryReader.class).in(Scopes.SINGLETON);
+
+          bind(ArtifactRepository.class)
+            .annotatedWith(Names.named(NOAUTH_ARTIFACT_REPO))
+            .to(DefaultArtifactRepository.class)
+            .in(Scopes.SINGLETON);
+          expose(ArtifactRepository.class).annotatedWith(Names.named(NOAUTH_ARTIFACT_REPO));
```

Comment:
you are anyway exposing these bindings, then you don't need to remove these in private module, in the private module, you can just have the ArtifactRepositoryReader binding.

Reply:
also is `ArtifactRepositoryReader` only used by ArtifactRepository? If some other classes wants to use it, will it fail to inject because it is private?

Reply:
I think we would need to have bind(RepoReader) and bind(Repo) in the same module otherwise RepoReader binding is not visible to bind(Repo). 

Yes. RepoReader is intended to be used by Repo only at this moment. 

Yes. If other classes need to use it. They would do something like 
PrivateModule {
  bind(RepoReader).to(...) 
  bind(SomeClassInterface).to(SomeClassImpl) 
}

Reply:
That is my concern since the purpose of binding is to make injection easier, a private module should only be used if a class is specific to some other classes. If some other classes needs an interface, they need to bind it themselves, that's no different than calling the constructor themselves.

Reply:
The use case here is using private module to allow different bindings for the same class. As in a follow-up CL, RepoReader class will be bound to local in appfabric and remote for preview when we use local level DB. 

Here is a stackoverflow thread related to this topic
https://stackoverflow.com/questions/16987815/guice-binding-several-objects-with-different-dependencies

In general, if we require class A to always be bound to class AImpl, then yes probably no need to use private module, or use private with expose. But if we need to bind class A to class AImplX or class AImplY depending on some other condition, then private module make the binding easy and cleaner to do.